### PR TITLE
feat(lobby): challenge-response messaging

### PR DESCRIPTION
**Pull request by doganbros**
Better moderation capabilities in locked down rooms: challenge/response in lobby [#9714](https://github.com/jitsi/jitsi-meet/issues/9714)
Messaging methods in lobby room
getLocalId to get the local room id from jid
setMessageListener to listen for messages in lobby room

Reviewed-by: Fecri Kaan Ulubey f.kaan93@gmail.com

### CODE REVIEW

File: JitsiConference.js [hunk 1]
```diff
@@ -3948,6 +3948,55 @@ JitsiConference.prototype.joinLobby = function(displayName, email) {
...
+};
+
+/**
+ * Set message listener for the lobby room
+ * @param {Function} listener The function to be called when a new message arrives
+ *
+ * @returns {void}
+ */
+JitsiConference.prototype.setLobbyMessageListener = function(listener) {
+    if (this.room) {
```
[the remark this answers is not in the document]
Comment (umitdogan1) [reply JitsiConference.js:3995]: Sure.

File: JitsiConference.js [hunk 2]
```diff
@@ -3948,6 +3948,55 @@ JitsiConference.prototype.joinLobby = function(displayName, email) {
...
+
+/**
+ * Sends a private message to a participant in a lobby room.
+ * @param {string} id The participant id.
+ * @param {Object} message The message to send
+ *
+ * @returns {void}
+ */
+JitsiConference.prototype.sendPrivateLobbyMessage = function(id, message) {
+    if (this.room) {
```
[the remark this answers is not in the document]
Comment (umitdogan1) [reply JitsiConference.js:3983]: Ok, will check.

File: JitsiConference.js [hunk 3]
```diff
@@ -3948,6 +3948,55 @@ JitsiConference.prototype.joinLobby = function(displayName, email) {
     return Promise.reject(new Error('The conference not started'));
 };
 
+/**
+ * Gets the local id for a participant in a lobby room.
+ * This is used for lobby room private chat messages.
+ *
+ * @returns {string}
```
[the remark this answers is not in the document]
Comment (umitdogan1) [reply JitsiConference.js:3955]: Yes more logical, on it...

**Comment by mkhstar** [comment]
@damencho we added a commit that addresses the issues you raised. We have also added a commit to jitsi-meet's pull request to sync the changes.

**Comment by damencho** [comment]
Jenkins, test this please.

**Comment by damencho** [comment]
Can you please rebase this one to latest so the PR testing can be executed, thank you.

**Comment by damencho** [comment]
Your call :) I will squash everything in the end ... so it doesn't matter for me :)

**Comment by mkhstar** [comment]
Let me squash them now

**Comment by mkhstar** [comment]
I just squashed them now. Thank you :)

**Comment by damencho** [comment]
Jenkins, test this please.

**Review by damencho — APPROVED** [review]

**Pull request merged**